### ISSUE TITLE
Pre-commit hook not supported with Python 3.14

### ISSUE THREAD

**Issue by frostbyte-ninja** [ansible/ansible-lint#4946]
Currently, it’s not possible to use the pre-commit hook with Python 3.14 because the Python version is pinned to 3.13:

```
pre-commit run --all-files
[INFO] Installing environment for https://github.com/ansible/ansible-lint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/python', '-mvirtualenv', '/home/user/.cache/pre-commit/repoekj7ow2v/py_env-python3.13', '-p', 'python3.13')
return code: 1
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.13'
stderr: (none)
Check the log at /home/user/.cache/pre-commit/pre-commit.log
```
